model-engine/model_engine_server/api/model_endpoints_v1.py

```diff
@@ -207,3 +209,27 @@ async def delete_model_endpoint(
             status_code=500,
             detail="deletion of endpoint failed, compute resources still exist.",
         ) from exc
+
+
+@model_endpoint_router_v1.post(
+    "/model-endpoints/{model_endpoint_id}/restart", response_model=RestartModelEndpointV1Response
+)
+async def restart_model_endpoint(
+    model_endpoint_id: str,
+    auth: User = Depends(verify_authentication),
+    external_interfaces: ExternalInterfaces = Depends(get_external_interfaces),
+) -> RestartModelEndpointV1Response:
+    """
+    Restarts the Model endpoint.
+    """
+    logger.info(f"POST /model-endpoints/{model_endpoint_id}/restart for {auth}")
+    try:
+        use_case = RestartModelEndpointV1UseCase(
+            model_endpoint_service=external_interfaces.model_endpoint_service,
+        )
+        return await use_case.execute(user=auth, model_endpoint_id=model_endpoint_id)
+    except (ObjectNotFoundException, ObjectNotAuthorizedException) as exc:
```

Comment:
we probably want to catch the ObjectHasInvalidValueException here and return 400 (or something)?